06_puppy/kasunfdo/syncstore.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+func (s *SyncStore) CreatePuppy(puppy *Puppy) bool {
+	_, exists := s.Map.LoadOrStore(puppy.ID, puppy)
+	// exists is true if a record exists with given key, false if value stored.
+	return !exists
+}
+
+func (s *SyncStore) ReadPuppy(id uint64) (*Puppy, bool) {
+	value, exists := s.Map.Load(id)
```

Comment:
This should be folded into the `if` statement as none of the variables are needed outside the body of the if:

```
if value, exists := s.Map.Load(id); exists {
        return value.(*Puppy), true
}
return nil, false
```